executedomainmanagerecstask_policy.tf

```diff
@@ -0,0 +1,146 @@
+
+data "aws_iam_policy_document" "executedomainmanagerecstask" {
+  statement {
+    actions = [
+      "acm:DeleteCertificate",
+      "acm:GetCertificate",
+      "acm:ListTagsForCertificate",
+      "acm:RemoveTagsFromCertificate",
+      "acm:UpdateCertificateOptions",
+    ]
+
+    resources = ["*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:ResourceTag/app"
+      values   = ["domain-manager"]
+    }
+  }
+
+  statement {
+    actions = [
+      "acm:AddTagsToCertificate",
+      "acm:DescribeCertificate",
+      "acm:ListCertificates",
+      "acm:RequestCertificate",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "cloudfront:DeleteDistribution",
+      "cloudfront:GetDistribution",
+      "cloudfront:GetDistributionConfig",
+      "cloudfront:UntagResource",
+      "cloudfront:UpdateDistribution"
+    ]
+
+    resources = ["*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:ResourceTag/app"
+      values   = ["domain-manager"]
+    }
+  }
+
+  statement {
+    actions = [
+      "cloudfront:CreateDistribution",
+      "cloudfront:CreateDistributionWithTags",
+      "cloudfront:ListDistributions",
+      "cloudfront:ListTagsForResource",
+      "cloudfront:TagResource",
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "cognito-idp:AdminAddUserToGroup",
+      "cognito-idp:AdminConfirmSignUp",
+      "cognito-idp:AdminCreateUser",
+      "cognito-idp:AdminDeleteUser",
+      "cognito-idp:AdminDisableUser",
+      "cognito-idp:AdminEnableUser",
+      "cognito-idp:AdminGetUser",
+      "cognito-idp:AdminInitiateAuth",
+      "cognito-idp:AdminListDevices",
+      "cognito-idp:AdminListGroupsForUser",
+      "cognito-idp:AdminRemoveUserFromGroup",
+      "cognito-idp:AdminResetUserPassword",
+      "cognito-idp:ListUsers",
+      "cognito-idp:ListUsersInGroup",
+      "cognito-idp:SignUp",
+    ]
+
+    resources = ["*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:ResourceTag/app"
+      values   = ["domain-manager"]
+    }
+  }
+
+  statement {
+    actions = [
+      "route53:ChangeResourceRecordSets",
+      "route53:ChangeTagsForResource",
+      "route53:CreateHostedZone",
+      "route53:DeleteHostedZone",
+      "route53:GetHostedZone",
+      "route53:ListHostedZones",
+      "route53:ListResourceRecordSets",
+      "route53:ListTagsForResource"
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "s3:CreateBucket",
+      "s3:DeleteBucket",
+      "s3:DeleteBucketPolicy",
+      "s3:DeleteBucketWebsite",
+      "s3:DeleteObject",
+      "s3:GetBucketWebsite",
+      "s3:GetObject",
+      "s3:ListAllMyBuckets",
+      "s3:ListBucket",
+      "s3:PutBucketPolicy",
+      "s3:PutBucketPublicAccessBlock",
+      "s3:PutBucketTagging",
+      "s3:PutBucketWebsite",
+      "s3:PutObject"
+    ]
+
+    resources = ["*"]
```

Comment:
@jsf9k Unfortunately, AWS does not allow tag-based access to buckets (https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazons3.html). The app needs to be able to generate s3 buckets with the same name as the domain for redirects as detailed by this https://aws.amazon.com/premiumsupport/knowledge-center/redirect-domain-route-53/.  I do not like these permissions either in a shared account, but there's not a way to avoid this, unless it's in a separate account.

Reply:
:disappointed:

Reply:
@zenine07 How much work would it be on the Domain Manager side if we hosted the buckets in a different account dedicated to Domain Manager?  Though if we go down that path, it might make sense to just create a separate account entirely for Domain Manager, like we did for Con-PCA.  Perhaps we were being too optimistic that DM would be a good first tenant in our User Services account?

@jsf9k and @zenine07 What do you think?

Reply:
@dav3r I agree, if we want the S3 buckets in a separate account, might as well put everything in a separate account at that point. Would probably be simpler.

Reply:
@jsf9k We can reuse a lot of the work we already have, so I don't think it will be too terrible.